Tests/External/Plugins/ASCompletion.Tests/MainForm.cs

```diff
@@ -0,0 +1,378 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Windows.Forms;
+using FlashDevelop.Utilities;
+using PluginCore;
+using ScintillaNet.Configuration;
+using WeifenLuo.WinFormsUI.Docking;
+
+namespace FlashDevelop
+{
+    public class MainForm : Form, IMainForm
```

Comment:
When you are unit testing, you should make it obvious that you are working with a mock object. We would tend to call something like this MockForm or MockMainForm and maybe drop it in a Mocks subdirectory of the project. This way, as people add additional tests, they can easily see what mocks exist and extend them as necessary.

The fact that you need to derive from Form is also a code smell. You should be able to test with just an IMainForm so that no UI is created. That may require refactoring your project so that you are never casting an IMainForm to a Form or passing a Form around. Any methods/properties on Form that you are using should be in the interface.

I realize this can be a difficult goal to achieve though. Unfortunately, Microsoft did not design WinForms for testability. If they did, the child controls would be something like IControl. You can get there by creating your own interfaces for everything, but it can be more work than it is worth.